### ISSUE TITLE
[BUG] Unhandled Exception: PlatformException(evaluateJavaScript_failed, Failed evaluating JavaScript, JavaScript string was: 'play()'

### ISSUE THREAD

**Issue by ManKeomorakort** [sarbagyastha/youtube_player_flutter#192]
**Describe the bug**
```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: PlatformException(evaluateJavaScript_failed, Failed evaluating JavaScript, JavaScript string was: 'play()'
Error Domain=WKErrorDomain Code=4 "A JavaScript exception occurred" UserInfo={WKJavaScriptExceptionLineNumber=1, WKJavaScriptExceptionMessage=ReferenceError: Can't find variable: play, WKJavaScriptExceptionColumnNumber=5, WKJavaScriptExceptionSourceURL=https://m.youtube.com/, NSLocalizedDescription=A JavaScript exception occurred})
#0      StandardMethodCodec.decodeEnvelope 
package:flutter/…/services/message_codecs.dart:569
#1      MethodChannel.invokeMethod 
package:flutter/…/services/platform_channel.dart:321
<asynchronous suspension>
#2      MethodChannelWebViewPlatform.evaluateJavascript 
package:webview_media/src/webview_method_channel.dart:113
#3      WebViewController.evaluateJavascript 
package:webview_media/webview_flutter.dart:684
#4      YoutubePlayerController._callMethod (package:youtube_player_flutter/src/ut<…>
```
**To Reproduce**
It happen when i play difference youtube videos in sort period. It mostly happen with android device such as OPPO and Iphone XR.

**Expected behavior**

**Screenshots**

**Technical Details:**
 - Device: [Iphone XR]
 - OS: [13.3]

**Additional context**
My environment:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale km-KH)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] Connected device (1 available)
```

**Comment by codepushr** [comment]
I got the same issue, but quite undeterministicly - it happens just from time to time and with different variables, I got the error with `mute` instead of `play`.

**Comment by ohjs88** [comment]
If you set the key to YoutubePlayer as below, no errors will occur.

```
final UniqueKey youtubeKey = UniqueKey();

_buildVideoView(BuildContext context) {
    return YoutubePlayer(
      key: youtubeKey,
      ...
    );
```

**Comment by ManKeomorakort** [comment]
> If you set the key to YoutubePlayer as below, no errors will occur.
> 
> ```
> final UniqueKey youtubeKey = UniqueKey();
> 
> _buildVideoView(BuildContext context) {
>     return YoutubePlayer(
>       key: youtubeKey,
>       ...
>     );
> ```
I am not sure this is going to work, cos i already use **key: ObjectKey(Uuid().v1())** and  i still got error on some devices.

**Comment by ohjs88** [comment]
> > If you set the key to YoutubePlayer as below, no errors will occur.
> > ```
> > final UniqueKey youtubeKey = UniqueKey();
> > 
> > _buildVideoView(BuildContext context) {
> >     return YoutubePlayer(
> >       key: youtubeKey,
> >       ...
> >     );
> > ```
> 
> I am not sure this is going to work, cos i already use **key: ObjectKey(Uuid().v1())** and i still got error on some devices.

You're right. As a result of today's test, if you modify it as below, it worked normally.

**raw_youtube_player.dart**
```
onPageFinished: (_) {
          if (_isPlayerReady) {
            controller.updateValue(
              controller.value.copyWith(isReady: true),
            );
          } 
          // Deactivate this segment.
          // else {
          //   controller.reload();
          // }
        },
```

If "onPageFinished" is called before the player is ready, it appears that the controller is reloaded, causing a problem.

If the problem I predict is correct, I want someone to commit.

**Comment by ManKeomorakort** [comment]
> > > If you set the key to YoutubePlayer as below, no errors will occur.
> > > ```
> > > final UniqueKey youtubeKey = UniqueKey();
> > > 
> > > _buildVideoView(BuildContext context) {
> > >     return YoutubePlayer(
> > >       key: youtubeKey,
> > >       ...
> > >     );
> > > ```
> > 
> > 
> > I am not sure this is going to work, cos i already use **key: ObjectKey(Uuid().v1())** and i still got error on some devices.
> 
> You're right. As a result of today's test, if you modify it as below, it worked normally.
> 
> **raw_youtube_player.dart**
> 
> ```
> onPageFinished: (_) {
>           if (_isPlayerReady) {
>             controller.updateValue(
>               controller.value.copyWith(isReady: true),
>             );
>           } 
>           // Deactivate this segment.
>           // else {
>           //   controller.reload();
>           // }
>         },
> ```
> 
> If "onPageFinished" is called before the player is ready, it appears that the controller is reloaded, causing a problem.
> 
> If the problem I predict is correct, I want someone to commit.

Ok I am waiting to test another version

**Comment by davidgalarza** [comment]
Any update, I really need to solve that?

**Comment by sarbagyastha** [comment]
Does this issue persists in v7.0.0 ?

**Comment by pdivita** [comment]
Yes I have, even using "final UniqueKey youtubeKey = UniqueKey();"
